scipy/sparse/linalg/matfuncs.py

```diff
@@ -592,6 +592,15 @@ def _expm(A, use_exact_onenorm):
     if len(A.shape) != 2 or A.shape[0] != A.shape[1]:
         raise ValueError('expected a square matrix')
 
+    # Trivial case
+    if A.shape == (1, 1):
+        data = np.array([[A.data[0]]])
+
+        # Sometimes underlying data dtype is
+        # not actually a numerical type
+        if np.issubdtype(data.dtype, np.number):
```

Comment:
Why do this? If something is passed in that `np.exp` can't handle it will throw.  Beyond that you're not doing anything with the failing case.  It just falls down through the existing code.